packages/flutter/lib/src/material/stepper.dart

```diff
@@ -689,10 +730,20 @@ class _StepperState extends State<Stepper> with TickerProviderStateMixin {
         ),
         if (!_isLast(i))
           Expanded(
-            child: Container(
-              margin: const EdgeInsets.symmetric(horizontal: 8.0),
-              height: 1.0,
-              color: Colors.grey.shade400,
+            child: Column(
+              children: <Widget>[
+                Container(
+                  margin: const EdgeInsets.symmetric(horizontal: 8.0),
+                  height: 1.0,
+                  color: Colors.grey.shade400,
+                ),
+                if (widget.type == StepperType.horizontal
+                    && widget.horizontalLinePosition == HorizontalLinePosition.top
+                    && widget.horizontalTitlePosition == HorizontalTitlePosition.bottom)
```

Comment:
It was confusing to me at first why adjusting [HorizontalLinePosition] had no effect, until I looked at the code and realized that [HorizontalTitlePosition] must also be [bottom]. I've suggested an update to the docstrings to help surface this.